### ISSUE TITLE
[Resource Card]: Headers are missing descriptive content

### ISSUE THREAD

**Issue by jendowns** [carbon-design-system/gatsby-theme-carbon#561]
## Detailed description

[Checkpoint 2.4.6 - Headings and Labels](http://www.ibm.com/able/guidelines/ci162/headings_and_labels.html)

You can see this on the demo site homepage, for example:
1. Go to `ResourceCard` component page: https://gatsby-theme-carbon.now.sh/components/ResourceCard
2. Run DAP tool with latest Sept 2019 ruleset
3. Observe violations for `Header missing descriptive content`:

![Screen Shot 2019-11-25 at 9 03 29 AM](https://user-images.githubusercontent.com/9057921/69552211-b1f4a280-0f63-11ea-83d9-d03807242b96.png)

This also affects the `FeatureCard`, which uses the `ResourceCard` (screenshot from demo site homepage):

![Screen Shot 2019-11-25 at 9 01 47 AM](https://user-images.githubusercontent.com/9057921/69552394-0435c380-0f64-11ea-8471-dc34782acfaf.png)

### Cause

This happens because when `title` or `subtitle` prop are not provided to the `ResourceCard` component, it proceeds to render empty heading elements anyway. These two props are not required.

**Comment by jendowns** [comment]
I will propose a PR to resolve this shortly 👍
